go/libraries/doltcore/schema/alterschema/modifycolumn.go

```diff
@@ -147,7 +147,7 @@ func updateTableWithModifiedColumn(ctx context.Context, tbl *doltdb.Table, oldSc
 		return nil, err
 	}
 	var autoVal types.Value
-	if schema.HasAutoIncrement(newSch) {
+	if schema.HasAutoIncrement(newSch) && schema.HasAutoIncrement(oldSch) {
```

Comment:
this looks much better, could you add a comment explaining the alter table case? Also, what happens if the `AUTO_INCREMENT` is removed from the schema?

Reply:
The new schema would not have auto increment an a nil value would be passed for autoVal